crates/router/src/core/payments.rs

```diff
@@ -76,6 +78,16 @@ where
         .validate_request(&req, &merchant_account)?;
 
     tracing::Span::current().record("payment_id", &format!("{}", validate_result.payment_id));
+
+    #[cfg(feature = "locking")]
```

Comment:
Are we sure, all APIs go through this flow?

Reply:
Afaik , most of the payments api go through this flow. The payments webhooks , and redirection definitely go through this flow.